examples/datastreams/main.go

```diff
@@ -0,0 +1,143 @@
+)
+
+func init() {
+	flag.StringVar(&host, "host", "", "livekit server host")
+	flag.StringVar(&apiKey, "api-key", "", "livekit api key")
+	flag.StringVar(&apiSecret, "api-secret", "", "livekit api secret")
+}
+
+func main() {
+	logger.InitFromConfig(&logger.Config{Level: "info"}, "ds-demo")
+	lksdk.SetLogger(logger.GetLogger())
+	flag.Parse()
+
+	roomName := "datastreams-demo-" + uuid.New().String()[:8]
+	senderRoom, err := lksdk.ConnectToRoom(host, lksdk.ConnectInfo{
+		APIKey:              apiKey,
+		APISecret:           apiSecret,
+		RoomName:            roomName,
+		ParticipantIdentity: sender,
+	}, nil)
+	if err != nil {
+		logger.Errorw("failed to connect to room", err, "participant", sender)
+		return
+	}
+
+	receiverRoom, err := lksdk.ConnectToRoom(host, lksdk.ConnectInfo{
+		APIKey:              apiKey,
+		APISecret:           apiSecret,
+		RoomName:            roomName,
+		ParticipantIdentity: receiver,
+	}, nil)
+	if err != nil {
+		logger.Errorw("failed to connect to room", err, "participant", receiver)
+		return
+	}
+
+	defer func() {
+		senderRoom.Disconnect()
+		receiverRoom.Disconnect()
+	}()
+
+	receiverRoom.RegisterTextStreamHandler("text-receive-iter", func(reader *lksdk.TextStreamReader, participantIdentity string) {
+		logger.Infow("received text stream", "participant", participantIdentity)
+		// Q. If I iterate over the reader directly without a goroutine, I don't receive any dc messages
+		go func() {
+			for chunk := range reader.Read() {
+				logger.Infow("received text chunk", "chunk", chunk)
+			}
+		}()
+	})
+
+	receiverRoom.RegisterTextStreamHandler("text-receive-all", func(reader *lksdk.TextStreamReader, participantIdentity string) {
+		logger.Infow("received text stream", "participant", participantIdentity)
+		go func() {
+			all := reader.ReadAll()
+			logger.Infow("received text", "text", all)
+		}()
+	})
+
+	// receiverRoom.RegisterTextStreamHandler("text-stream-recv-iter", func(reader *lksdk.TextStreamReader, participantIdentity string) {
+	// 	logger.Infow("received text stream", "participant", participantIdentity)
+	// 	go func() {
+	// 		for chunk := range reader.Read() {
+	// 			logger.Infow("received text chunk", "chunk", chunk)
+	// 		}
+	// 	}()
+	// })
+
+	// receiverRoom.RegisterTextStreamHandler("text-stream-recv-all", func(reader *lksdk.TextStreamReader, participantIdentity string) {
+	// 	logger.Infow("received text stream", "participant", participantIdentity)
+	// 	go func() {
+	// 		all := reader.ReadAll()
+	// 		logger.Infow("received text", "text", all)
+	// 	}()
+	// })
+
+	// receiverRoom.RegisterByteStreamHandler("file-receive", func(reader *lksdk.ByteStreamReader, participantIdentity string) {
+	// 	logger.Infow("received file stream", "participant", participantIdentity)
+	// 	go func() {
+	// 		all := reader.ReadAll()
+	// 		os.WriteFile("received.pdf", all, 0644)
+	// 	}()
+	// })
+
+	// Q. If I don't add a delay, I don't receive stream header, I do get chunk and trailer though
```

Comment:
Without a ~100ms delay here, I somehow don't get a header, but chunks and trailer arrive as expected.

Reply:
did you test this with a local livekit server or against cloud?

Reply:
cloud (prod, mumbai)

Reply:
Is this consistently reproducible?

Reply:
yep it is fairly consistent

Reply:
It was due to DC not being ready on the receiving side